pkg/dockerregistry/server/writelimiter/blobstore.go

```diff
@@ -0,0 +1,106 @@
+package writelimiter
+
+import (
+	"runtime"
+	"sync/atomic"
+
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/context"
+
+	"github.com/openshift/origin/pkg/dockerregistry/server/types"
+)
+
+type blobWriter struct {
+	distribution.BlobWriter
+	release func()
+}
+
+var _ distribution.BlobWriter = blobWriter{}
+
+func (bw blobWriter) Close() error {
+	bw.release()
+	return bw.BlobWriter.Close()
+}
+
+type blobStore struct {
+	distribution.BlobStore
```

Comment:
`Close()` must be called any way on the writer if it had been initialized. Nevertheless, if we release in these methods as well, we can rely less on the finalizer.